### PR TITLE
[CSI] Install by default when Portworx is enabled

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <ggriffiths@purestorage.com>

**What this PR does / why we need it**:
* Installs CSI by default unless explicitly disabled.
* Fixes many unit tests that assumed CSI was disabled by default.

**Which issue(s) this PR fixes** (optional)
n/a

**Special notes for your reviewer**:
n/a

### CODE REVIEW

File: drivers/storage/portworx/component/csi.go [hunk 1]
```diff
@@ -72,14 +72,18 @@ func (c *csi) Initialize(
...
-	return pxutil.FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
+	return pxutil.IsCSIEnabled(cluster)
 }
 
 func (c *csi) Reconcile(cluster *corev1.StorageCluster) error {
 	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
 	pxVersion := pxutil.GetPortworxVersion(cluster)
 	csiConfig := c.getCSIConfiguration(cluster, pxVersion)
 
+	if cluster.Status.DesiredImages == nil {
```
Comment (ggriffiths) [inline drivers/storage/portworx/component/csi.go:83]: This normally won't be called, as we usually set defaults in another function. However, for tests, this is needed.

File: drivers/storage/portworx/util/util.go [hunk 2]
```diff
@@ -216,6 +216,18 @@ func IsPortworxEnabled(cluster *corev1.StorageCluster) bool {
...
+// IsCSIEnabled returns true if CSI is not disabled by the feature flag
+func IsCSIEnabled(cluster *corev1.StorageCluster) bool {
+	enableCSI := true
+	if FeatureCSI.IsSet(cluster.Spec.FeatureGates) {
+		enableCSI = FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
+	}
+
+	// Only enable CSI when Portworx is enabled
+	enableCSI = IsPortworxEnabled(cluster) && enableCSI
+	return enableCSI
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/util/util.go:228]: nit: The `feature.IsEnabled` method in feature.go could be simply written like this - 
```
func (feature Feature) IsEnabled(featureMap map[string]string) bool {
	enabled, err := strconv.ParseBool(featureMap[string(feature)])
	return err != nil || enabled
}
```
The above also makes sure, if the user has not properly disabled (by giving non-boolean or empty value) the feature, then we still return true.

Then we can simply write this helped method as follows - 
```
return IsPortworxEnabled(cluster) && FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
```

**Review by piyush-nimbalkar — APPROVED** [review]
Are we planning to make CSI default from operator 2.0.0 and px 3.0.0? Is there are ticket for this in Operator jira? If not, can we create one for tracking?

File: drivers/storage/portworx/util/feature.go [hunk 3]
```diff
@@ -12,8 +12,13 @@ const (
...
 func (feature Feature) IsEnabled(featureMap map[string]string) bool {
 	enabled, err := strconv.ParseBool(featureMap[string(feature)])
-	return err == nil && enabled
+	return err != nil || enabled
+}
+
+// IsSet checks if the feature is set in the given feature map
+func (feature Feature) IsSet(featureMap map[string]string) bool {
+	_, ok := featureMap[string(feature)]
+	return ok
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/util/feature.go:23]: We don't need this now.

**Review by piyush-nimbalkar — APPROVED** [review]

**Review by piyush-nimbalkar — APPROVED** [review]

**Pull request merged**